src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/CollectionApiDataType.java

```diff
@@ -2,8 +2,12 @@
 
 import com.datastax.oss.driver.api.core.type.DataType;
 
-/** Interface defining the api data type for collection types */
-public abstract class CollectionApiDataType implements ApiDataType {
+/**
+ * Interface defining the api data type for collection types. <br>
+ * DataType T is a generic type that can be map/set/list/vector, inheritors should implement
```

Comment:
use the @param doc for T - we are making this generic because the driver has some properties only available on the leaf level data types